styleguide/drafts/tooling.rst

```diff
@@ -0,0 +1,518 @@
+
+  * Running Test, Build is the same in "every"  project e.g. `nox -s test`
+  * Developers are more likely to develop deep knowledge of the tools
+
+* Reduces the potential of errors and misuse
+* Increases potential for automation (checks, automated fixes, etc.)
+* Decreases complexity of automation
+
+Overview
+________
+
+Task Runner & Automation
+++++++++++++++++++++++++
+
+* Nox_       (task runner, build automation, test & build isolation)
+* PreCommit_ (commit hooks)
+
+Code Style
+++++++++++
+
+* Black_     (style, code formatter)
+* Isort_     (style, import sorting)
+
+Testing
++++++++
+
+* Pytest_    (ut, testing)
+
+Documentation
++++++++++++++
+
+* Sphinx_    (documentation)
+
+Dependencies & Packaging
+++++++++++++++++++++++++
+
+* Poetry_    (dependency management, virtual environment, build-tool, packaging)
+
+Code Health
++++++++++++
+
+* Pylint_    (linter, code analysis)
+* Coverage_  (coverage)
+* MyPy_      (linter, type checking)
+* Sonar_     (linter, code coverage, code health)
+
+
+Up For Consideration
+++++++++++++++++++++
+
+* `scriv <https://github.com/nedbat/scriv>`_ (changelog management)
+* `prysk <https://www.prysk.net/>`_ (Integration testing cli's and co.)
+
+
+Task Runner & Automation
+________________________
+
+A task runner is great for automating common workflows like running tests, building the project etc..
+A second benefit of a task runner is, that it is a "in code"/"in config" documentation of the steps
+required for a specific task. If common task also get the same runner names across
+projects, switching between projects is simplified big time.
+
+.. figure:: https://imgs.xkcd.com/comics/sandwich.png
+    :alt: xkcd-149
+    :target: https://xkcd.com/149/
+
+    source: xkcd_
+    license: `CC BY-NC 2.5`_
+
+
+Nox_
+++++
+`Project Homepage <Nox_www_>`_
+
+Pros & Cons
+~~~~~~~~~~~
+
+.. list-table::
+    :header-rows: 1
+
+    * - Pros ✅
+      - Cons ❌
+    * - Python based (only requires python which is a prerequisite for a python project)
+      - Parametrized targets could be more convenient
+    * - Defined as code (missing functionality easily can be added)
+      - Still assumes PIP as basic package manager
+    * - Platform dependent
+      -
+    * - Overall functionality is small, therefore easy to learn
+      -
+
+Other Projects using Nox
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+* `pip <https://github.com/pypa/pip>`_
+* `pipx <https://github.com/pypa/pipx>`_
+* `Salt <https://github.com/saltstack/salt>`_
+* `Urllib3 <https://github.com/urllib3/urllib3>`_
+* `Hydra <https://hydra.cc/>`_
+
+Alternatives
+~~~~~~~~~~~~
+
+* `tox <https://tox.wiki/en/latest/>`_
+* `make <https://en.wikipedia.org/wiki/Make_(software)>`_
+* `cargo-make <https://sagiegurari.github.io/cargo-make/>`_
+* `just <https://github.com/casey/just>`_
+* `Ant <https://ant.apache.org/>`_
+* `Task <https://taskfile.dev/#/>`_
+* `Tusk <https://github.com/rliebz/tusk>`_
+* `Gulp <https://gulpjs.com/>`_
+* `Grunt <https://gruntjs.com/>`_
+
```

Comment:
We should mention https://github.com/nat-n/poethepoet, because we still use it in some projects